microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/ProjectTypeRegistry.java

```diff
@@ -0,0 +1,106 @@
+package com.redhat.microprofile.jdt.internal.core;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IProjectDescription;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.IExtensionRegistry;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.ls.core.internal.ProjectUtils;
+
+import com.redhat.microprofile.jdt.core.IProjectTypeIdentifier;
+import com.redhat.microprofile.jdt.core.MicroProfileCorePlugin;
+
+public class ProjectTypeRegistry {
+
+	private static final String CLASS_ATTR = "class";
+
+    private static final String ID = "projectType";
+    
+    private static final String MAVEN_TYPE = "maven";
+    
+    private static final String GRADLE_TYPE = "gradle";
+
+    private static final ProjectTypeRegistry INSTANCE = new ProjectTypeRegistry();
+   
+    private final Map<IProjectTypeIdentifier, String /* project type */> identifiers;
```

Comment:
I think a simple list should be enough. I suggest you create a new class ProjectTypeDefinition which store label, description (coming from attributes of the extension point) and store the IProjectType instance coming from class